compiler/qsc_hir/src/hir.rs

```diff
@@ -1361,6 +1361,25 @@ pub enum Attr {
     Reset,
 }
 
+impl Attr {
+    /// Gets the string description of the attribute.
+    #[must_use]
+    pub fn description(&self) -> &'static str {
+        match self {
+            Attr::Config => "Provides pre-processing information about when an item should be included in compilation.
+
+Valid arguments are `Base`, `Adaptive`, `IntegerComputations`, `FloatingPointComputations`, `BackwardsBranching`, `HigherLevelConstructs`, `QubitReset`, and `Unrestricted`.
+
+The `not` operator is also supported to negate the attribute, e.g. `not Adaptive`.",
+            Attr::EntryPoint => "Indicates that a callable is an entry point to a program.",
```

Comment:
nitpick: since we only allow a single entry point, this should be "the"
```suggestion
            Attr::EntryPoint => "Indicates that the callable is the entry point to a program.",
```